control-plane/connect-inject/peering_acceptor_controller_test.go

```diff
@@ -459,8 +454,24 @@ func TestReconcileDeletePeeringAcceptor(t *testing.T) {
 			// Add the default namespace.
 			ns := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}
 
-			// Create fake k8s client.
-			k8sObjects := []runtime.Object{&ns}
+			acceptor := &v1alpha1.PeeringAcceptor{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "acceptor-deleted",
+					Namespace:         "default",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+					Finalizers:        []string{FinalizerName},
+				},
+				Spec: v1alpha1.PeeringAcceptorSpec{
+					Peer: &v1alpha1.Peer{
+						Secret: &v1alpha1.Secret{
+							Name:    "acceptor-deleted-secret",
+							Key:     "data",
+							Backend: "kubernetes",
+						},
+					},
+				},
+			}
+			k8sObjects := []runtime.Object{&ns, acceptor}
 
```

Comment:
Would it make sense to assert that the Acceptor's finalizer is removed at the bottom of this test? Since after calling reconcile we'd expect that to be the case? I guess in real K8s env, the resource is probably being deleted but in this test we're just calling reconcile so we might be able to make the assertion that the finalizer is removed.